NEWS.md

```diff
@@ -5,6 +5,41 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.10.0] – unreleased
+
+### Changed
+
+* `Distributions.jl`, `RecursiveArrayTools.jl` and `HybridArrays.jl` were moved to weak dependencies to reduce load time and improve extensibility.
```

Comment:
We could maybe mention what this implies in practice? Everyone who until now uses the `ArrayPartition` (probably everyone using product manifold until now) has to load `RecursiveArrayTools` manually now. 

Where does this change things for the other two?

Reply:
Practical implications are already discussed in the "Removed" section, do you think it's insufficient?

Reply:
Removed just says which ones are no longer removed – either there or here we should then mention “where to get them from instead”

Reply:
I've added some more details. For `HybridArrays.jl` there are no practical implications, and for `Distributions.jl` its mostly "if you have to use it, it's still there but private; we would like to have a proper exported replacement in the future but other projects take priority"